### PR TITLE
switch to github actions for building jekyll

### PR DESCRIPTION
**Pull request by jesopo**
this introduces two workflows
- a workflow to build the jekyll site
- a workflow to deploy the site to GitHub Pages

the former runs on every branch, the latter only runs on `main`

this PR also upgrades us to jekyll `4.3.0`, and means that every PR will have jekyll config changes tested

### CODE REVIEW

File: .github/workflows/jekyll-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Deploy Jekyll site to Pages
+
+on:
+  workflow_run:
+    workflows: [Build Jekyll site]
+    branches: [main, jess/actions-test]
```
Comment (jesopo) [inline .github/workflows/jekyll-deploy.yml:6]: `jess/actions-test` ought to be removed when this PR is ready to be merged

File: .github/workflows/jekyll-build.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Build Jekyll site
+
+on: [push, pull_request]
```
Comment (swantzter) [inline .github/workflows/jekyll-build.yml:3]: I'd consider also allowing workflow_dispatch

File: .github/workflows/jekyll-build.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+permissions:
```
Comment (swantzter) [inline .github/workflows/jekyll-build.yml:5]: do you need custom permissions for just the build part?

File: .github/workflows/jekyll-build.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/checkout@v3
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@0a29871fe2b0200a17a4497bae54fe5df0d973aa # v1.115.3
+        with:
+          ruby-version: '3.0' # Not needed with a .ruby-version file
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+          cache-version: 0 # Increment this number if you need to re-download cached gems
+      - name: Build with Jekyll
+        # Outputs to the './_site' directory by default
+        run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
```
Comment (swantzter) [inline .github/workflows/jekyll-build.yml:23]: steps.pages doesn't exist.
Move
```yml
      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@v2
```
from `jekyll-deploy.yml` here, that step is used to configure jekyll/other before the build?

File: .github/workflows/jekyll-build.yml (hunk 3 above)
Comment (jesopo) [reply .github/workflows/jekyll-build.yml:5]: I think we need some permissions to upload the artifact, no?

Comment (swantzter) [reply .github/workflows/jekyll-build.yml:5]: Sure, but I think that's included in the default ones
id-token isn't default write and I assume you need that to do the actual deploy, but I don't think you need it to build

**Pull request merged**